bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -108,7 +108,7 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
         logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
+        if (!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {
```

Comment:
`isEcho` returns boolean but has the side effect of removing the ignoreEvent from the list.  With this change, there are cases where `isEcho` is no longer reached, likely leading to a memory leak and other problems when the manual switch is not set.

I would like to learn more about how there is a fundamental difference between the 1.x runtime, and the compat1x layer, in sending events.  How did you discover this difference, @lewie?  Also, is there a way to avoid requiring manual configuration of something that ought to be learnable at runtime without configuration?